enclave-runtime/src/lib.rs

```diff
@@ -451,9 +452,26 @@ pub unsafe extern "C" fn sync_parentchain(
 /// * sends `confirm_call` xt's of the executed unshielding calls
 /// * sends `confirm_blocks` xt's for every synced parentchain block
 fn sync_parentchain_internal(blocks_to_sync: Vec<SignedBlock>) -> Result<()> {
-	let block_import_dispatcher = GLOBAL_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()?;
-
-	block_import_dispatcher.dispatch_import(blocks_to_sync).map_err(|e| e.into())
+	if let Ok(block_import_dispatcher) =
+		GLOBAL_TRIGGERED_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()
+	{
+		block_import_dispatcher.dispatch_import(blocks_to_sync)?;
+	} else if let Ok(block_import_dispatcher) =
+		GLOBAL_IMMEDIATE_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()
+	{
```

Comment:
I get your approach here. However, I think we can make the semantics here a bit more clear if we write it like this:

```rust
let block_import_dispatcher = match WORKER_MODE {
    WorkerMode::Sidechain => GLOBAL_TRIGGERED_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()?,
    _ => GLOBAL_IMMEDIATE_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()?,
};

```

Reply:
The issue is, that those types are not the same, I cannot assign two different types to the same variable. If they were the same, I wouldn't need to have two instances of the component container.

Reply:
Ah yes obviously. Then let me give another idea :D

```rust

match Worker_Mode {
     WorkerMode::Sidechain => GLOBAL_TRIGGERED_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()?.dispatch_import(...)?,
     _ => GLOBAL_IMMEDIATE_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT.get()?.dispatch_import(..)?,
};
```

But I have no strong opinion. I leave it to you if you want to implement that.

Reply:
Alright, yes, I will simplify it according to your suggestion 👍